lib/coinbase/staking_operation.rb

```diff
@@ -222,8 +252,22 @@ def stake_api
     def from_model(model)
       @model = model
       @status = model.status
-      @transactions = model.transactions.map do |transaction_model|
-        Transaction.new(transaction_model)
+      @transactions ||= []
```

Comment:
We should never be calling `from_model` in a world where these are already set? Or if we are, can we call this something like `reset_model` or something like that?

Also, we really probably only need `update_transactions(model)` since `@model = model` covers our bases since `@status` should not really be needed

Reply:
Staking operation `status` being complete means something very specific and is complete controlled server side. So for long running operations like for dedicated eth staking, we keep calling reload which helps fetch the latest staking operation that will then help update both txs and status. So we need a helper to do both.

Reply:
> So we need a helper to do both.

Sorry, I just mean that `@status` is not used:

https://github.com/coinbase/coinbase-sdk-ruby/blob/dac8daf899e524c65529c31a938f1c891192cc8b/lib/coinbase/staking_operation.rb#L109-L111

Instead of

https://github.com/coinbase/coinbase-sdk-ruby/blob/dac8daf899e524c65529c31a938f1c891192cc8b/lib/coinbase/staking_operation.rb#L79-L80C1

I would do something more targeted like:

```
    def initialize(model)
      @model = model
      update_transactions(model.transactions)
    end
```

```
    def reload
      @model = Coinbase.call_api do
        if wallet_id.nil?
          stake_api.get_external_staking_operation(network.id, address_id, id)
        else
          stake_api.get_staking_operation(wallet_id, address_id, id)
        end
      end

      update_transactions(@model.transactions)
    end
```

Reply:
@drohit-cb does that make sense?

Reply:
Done 👍